jooby/src/main/java/io/jooby/spi/ValueContainer.java

```diff
@@ -0,0 +1,80 @@
+package io.jooby.spi;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import io.jooby.Value;
+
+/**
+ *  A restricted base type of {@link Value} for the {@link ValueConverter} SPI.
+ * @author agentgt
+ *
+ */
+public interface ValueContainer {
```

Comment:
Mainly to avoid recursive ‘Value.in’.

Also because I’m not a fan of interfaces in extension APIs having an enormous number of methods (rxjava comes to mind). Also some people like to mock when unit testing thus smaller interfaces are easier.

Reply:
Sorry I mean recursive  ‘Value.to’ and not Value.in.

For example do we really want plugins to be able to depend on other plugins?

Also we might want to limit what is allowed for conversion. Maybe only isObject is supported.

And finally you might notice the new spi package is completely decoupled from all other Jooby things (the package can compile in isolation).